include/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.hpp

```diff
@@ -168,14 +168,17 @@ class Propagator
 
     /// @brief              Calculate the state subject to an Event Condition, given initial state and maximum end time
     /// @code
-    ///                     State state = propagator.calculateStateAt(aState, anInstant, anEventCondition);
+    ///                     NumericalSolver::ConditionSolution state = propagator.calculateStateAt(aState, anInstant,
+    ///                     anEventCondition);
     /// @endcode
     /// @param              [in] aState An initial state
     /// @param              [in] anInstant An instant
     /// @param              [in] anEventCondition An event condition
-    /// @return             State
+    /// @return             NumericalSolver::ConditionSolution
 
-    State calculateStateAt(const State& aState, const Instant& anInstant, const EventCondition& anEventCondition) const;
+    NumericalSolver::ConditionSolution calculateStateAt(
```

Comment:
I think it could be confusing to have two functions `calculateStateAt`/`calculateStatesAt` that do indeed return States and one `calculateStateAt` that does not.

What do you think of either making a separate function or renaming the current function to something along the lines of `propagateStateToCondition` to use for this case?